api/v1/atlasintegration_types.go

```diff
@@ -0,0 +1,187 @@
+}
+
+type DatadogIntegration struct {
+	// APIKeySecret is the name of a secret containing the datadog api key
+	// +kubebuilder:validation:Required
+	APIKeySecret string `json:"apiKeySecret"`
+
+	// Region is the Datadog region
+	// +kubebuilder:validation:Required
+	Region string `json:"region"`
+
+	// SendCollectionLatencyMetrics flags whether or not to send collection latency metrics
+	// +kubebuilder:validation:Required
+	SendCollectionLatencyMetrics bool `json:"sendCollectionLatencyMetrics"`
+
+	// SendDatabaseMetrics flags whether or not to send database metrics
+	// +kubebuilder:validation:Required
+	SendDatabaseMetrics bool `json:"sendDatabaseMetrics"`
+}
+
+type MicrosoftTeamsIntegration struct {
+	// URLSecret is the name of a secret containing the microsoft teams secret URL
+	// +kubebuilder:validation:Required
+	URLSecret string `json:"apiKeySecret"`
+}
+
+type NewRelicIntegration struct {
+	// CredentialsSecret is the name of a secret containing new relic's credentials:
+	// account id, license key, read and write tokens
+	// +kubebuilder:validation:Required
+	CredentialsSecret string `json:"credentialsSecret"`
+}
+
+type OpsGenieIntegration struct {
+	// APIKeySecret is the name of a secret containing Ops Genie's API key
+	// +kubebuilder:validation:Required
+	APIKeySecret string `json:"apiKeySecret"`
+
+	// Region is the Ops Genie region
+	// +kubebuilder:validation:Required
+	Region string `json:"region"`
+}
+
+type PagerDutyIntegration struct {
+	// ServiceKeySecret is the name of a secret containing Pager Duty service key
+	// +kubebuilder:validation:Required
+	ServiceKeySecret string `json:"serviceKeySecret"`
+
+	// Region is the Pager Duty region
+	// +kubebuilder:validation:Required
+	Region string `json:"region"`
+}
+
+type PrometheusIntegration struct {
+	// UsernameSecret is the name of a secret containing the Prometehus username
+	// +kubebuilder:validation:Required
+	UsernameSecret string `json:"usernameSecret"`
+
+	// ServiceDiscovery to be used by Prometheus
+	// +kubebuilder:validation:Enum:=file;http
+	// +kubebuilder:validation:Required
+	ServiceDiscovery string `json:"region"`
+
+	// Enabled flags whether or not Prometheus integration is enabled
+	// +kubebuilder:validation:Required
+	Enabled bool `json:"sendCollectionLatencyMetrics"`
+}
+
+type SlackIntegration struct {
+	// APITokenSecret is the name of a secret containing the Slack API token
+	// +kubebuilder:validation:Required
+	APITokenSecret string `json:"usernameSecret"`
+
+	// ChannelName to be used by Prometheus
+	// +kubebuilder:validation:Required
+	ChannelName string `json:"channelName"`
+
+	// TeamName flags whether or not Prometheus integration is enabled
+	// +kubebuilder:validation:Required
+	TeamName string `json:"teamName"`
+}
+
+type VictorOpsIntegration struct {
+	// KeysSecret is the name of a secret containing Victor Ops API and routing keys
+	// +kubebuilder:validation:Required
+	KeysSecret string `json:"keySecret"`
```

Comment:
routing key is not a secret (not redacted)

Reply:
will take it out, thx!

Reply:
Done